src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -1203,23 +1223,49 @@ Overlay::Setup
 setup_Overlay (BasicConfig const& config)
 {
     Overlay::Setup setup;
-    auto const& section = config.section("overlay");
-    setup.context = make_SSLContext("");
-    setup.expire = get<bool>(section, "expire", false);
+    {
+        auto const& section = config.section("overlay");
+        setup.context = make_SSLContext("");
+        setup.expire = get<bool>(section, "expire", false);
 
-    set (setup.ipLimit, "ip_limit", section);
-    if (setup.ipLimit < 0)
-        Throw<std::runtime_error> ("Configured IP limit is invalid");
+        set(setup.ipLimit, "ip_limit", section);
+        if (setup.ipLimit < 0)
+            Throw<std::runtime_error>("Configured IP limit is invalid");
 
-    std::string ip;
-    set (ip, "public_ip", section);
-    if (! ip.empty ())
+        std::string ip;
+        set(ip, "public_ip", section);
+        if (!ip.empty())
+        {
+            boost::system::error_code ec;
+            setup.public_ip = beast::IP::Address::from_string(ip, ec);
+            if (ec || beast::IP::is_private(setup.public_ip))
+                Throw<std::runtime_error>("Configured public IP is invalid");
+        }
+    }
     {
-        boost::system::error_code ec;
-        setup.public_ip = beast::IP::Address::from_string (ip, ec);
-        if (ec || beast::IP::is_private (setup.public_ip))
-            Throw<std::runtime_error> ("Configured public IP is invalid");
+        auto const& section = config.section("crawl");
+        bool crawlEnabled = true;
+
+        if (section.lines().size() == 1 && section.legacy() == "off") {
```

Comment:
A couple comments here:
1) If there's an error in the config file, it's silently ignored here. It also enables crawl on error, which isn't a good default. I'd probably throw here and not allow rippled to run with an invalid config file. But at a minimum I'd disable crawl on error and write to the log.
2) I wouldn't call the value "off". Since it's a boolean, I'd use 0/1 instead.
3) I wouldn't use the `legacy` function in new code. I'd use `values` instead.

Reply:
1.) the old default was to enable crawl, so I tried to keep it that way (i.e. always try to enable it unless explicitly disabled), but I've reworked it a bit.  
2.) I prefer "off" in this case because it's not one of the key/value pairs, it's just a single value, and seems clearer with the text. The example config is updated with listing "off" as the only option, it's not listed as a `<flag>`.
```
[crawl]
off
``` 
vs
```
[crawl]
0
```
3.) That works when adding more error handling. It wasn't clear upon first pass what happens if I throw an exception here, but the code above it does Throw for errors in the config file. I just learned that you don't see those exceptions in the debug-log or stdout by default though.

Reply:
For 1) I'm OK with default on, but not on error.

For 2) I don't want to introduce the new vocabulary words of `off` and `on` unless all the sections support it. 

Other configuration sections use `0`. For example:
```
[ssl_verify]
0
```
It's not consitant that we can say `[crawl] off` and not `[ssl_verify] off`. Also, if you really do want to keep `off`, I'd explictly check for `on`/`off`/`0`/`1`/`true`/`false` and signal an error if it's not in that set. I.e. `[crawl] o` should be flagged as an error. But to be clear, I vote for `0/1` unless we make this change for all sections.

For 3) I did not know that, I thought we did log errors. At any rate, I think that's an issue for a different PR. I do think we should throw on an error in the config file. We shouldn't let a misconfigured rippled run.

Reply:
That's fair, updated.